File: DoitMission_11/Doit11_Java/app/src/main/java/com/example/doit11_java/TextService.java
package com.example.doit11_java;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.Log;

public class TextService extends Service {


    public TextService() {
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {

        Log.e("Start", "onStartComman() 함수 호출");

        if(intent == null){
            return Service.START_STICKY;
        } else{
            processIntent(intent);
        }

        return super.onStartCommand(intent, flags, startId);
    }

    public void processIntent(Intent intent){

        String command = intent.getStringExtra(MainActivity.name);
        String message = intent.getStringExtra(MainActivity.message);

        Log.e("Start",command + " : " + message + " 서비스 호출");

        /*try{
            Thread.sleep(5000);
            Log.e("Start","Thread 호출중");
        }catch (Exception o){
            o.printStackTrace();
        }*/

        Intent showIntent = new Intent(getApplicationContext(),MainActivity.class);

        showIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        showIntent.putExtra(MainActivity.name,command);
        showIntent.putExtra(MainActivity.message,message);
        startActivity(showIntent);

    }

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public IBinder onBind(Intent intent) {
        throw new UnsupportedOperationException("Not yet implemented");
    }
}

File: DoitMission_11/Doit11_Kotlin/app/src/main/java/com/example/doit11_kotlin/TextService.kt
package com.example.doit11_kotlin

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.Log

class TextService : Service() {

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        if(intent == null) {
            return START_STICKY
        } else {
            process1(intent)
        }
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent?): IBinder? {
        throw UnsupportedOperationException("Not yet implemented")
    }

    private fun process1(intent : Intent){
        val name = intent.getStringExtra(MainActivity.name)
        val message = intent.getStringExtra(MainActivity.message)

        Log.e("Start", "$name + 호출 : $message")

        val backIntent = Intent(applicationContext,MainActivity::class.java)
        backIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
        backIntent.putExtra(MainActivity.name,name)
        backIntent.putExtra(MainActivity.message,message)
        startActivity(backIntent)
    }
}
